azure-container-apps-maven-plugin/src/main/java/com/microsoft/azure/maven/containerapps/ConfigMojo.java

```diff
@@ -0,0 +1,432 @@
+            .forSubscription(getSubscriptionId()).listSupportedRegions("Container App");
+        assert CollectionUtils.isNotEmpty(regions) : "No valid region found for Container App.";
+        this.wrapper.putCommonVariable("regions", regions);
+        final Region defaultRegion = regions.contains(Region.US_EAST) ? Region.US_EAST : regions.get(0);
+        final Region result = this.wrapper.handleSelectOne("configure-region", regions, defaultRegion, Region::getName);
+        this.region = result.getName();
+    }
+
+    private void configureAppName() throws IOException, InvalidConfigurationException {
+        final String appName = this.wrapper.handle("configure-app-name", false);
+        this.appName = appName;
+        this.wrapper.putCommonVariable("appName", appName);
+    }
+
+    private void configCommon() throws IOException, InvalidConfigurationException {
+        configureRegistry();
+        configureIdentity();
+        configureContainers();
+        configureIngress();
+        configureScale();
+    }
+
+    private void configureRegistry() throws IOException, InvalidConfigurationException {
+        final boolean useExistingRegistry = this.wrapper.handleConfirm("Use existing Azure Container Registry (Y/n):", true, true);
+        if (useExistingRegistry) {
+            selectRegistry();
+        } else {
+            final String registryName = this.wrapper.handle("configure-registry-name", false);
+            if (StringUtils.isNotBlank(registryName)) {
+                this.registry = new ContainerRegistryConfig();
+                this.registry.setRegistryName(registryName);
+            }
+        }
+        this.wrapper.putCommonVariable("registry", this.registry);
+    }
+
+    private void selectRegistry() throws IOException, InvalidConfigurationException {
+        log.info("It may take a few minutes to list Azure Container Registries in your account, please be patient.");
+        final AzureContainerRegistryModule module = Azure.az(AzureContainerRegistry.class).registry(subscriptionId);
+        final List<ContainerRegistry> registries = module.list();
+        this.wrapper.putCommonVariable("registries", registries);
+        final ContainerRegistry targetRegistry = this.wrapper.handleSelectOne("select-registry", registries, null, AbstractAzResource::getName);
+        if (targetRegistry != null) {
+            log.info(String.format("Using Azure Container Registry: %s", TextUtils.blue(targetRegistry.getName())));
+        }
+        this.registry = new ContainerRegistryConfig();
+        this.registry.setRegistryName(targetRegistry.getName());
+        this.registry.setResourceGroup(targetRegistry.getResourceGroup().getName());
+    }
+
+    private void configureIdentity() throws IOException, InvalidConfigurationException {
+        final boolean useExistingIdentity = this.wrapper.handleConfirm("Use existing User Assigned Identity (y/N):", false, true);
+        if (useExistingIdentity) {
```

Comment:
no parameter is needed if create a new identity? Do we just uses managed identity in this case? Shall we show a info here?

Reply:
Here we use the managed identity for image pulling. But we cannot assign ACR_PULL role for the system assigned identity before it is created. So we suggest use user assigned identtiy (Here we prompt users to select a existing user assigned identity) or use the system assigned identity only if it is already created.